ml-agents/mlagents/trainers/common/tf_optimizer.py

```diff
@@ -0,0 +1,156 @@
+from typing import Dict, Any, List, Tuple, Optional
+import numpy as np
+
+from mlagents.tf_utils.tf import tf
+from mlagents.trainers.buffer import AgentBuffer
+from mlagents.trainers.tf_policy import TFPolicy
+from mlagents.trainers.common.optimizer import Optimizer
+from mlagents.trainers.trajectory import SplitObservations
+from mlagents.trainers.components.reward_signals.reward_signal_factory import (
+    create_reward_signal,
+)
+from mlagents.trainers.components.bc.module import BCModule
+
+
+class TFOptimizer(Optimizer):  # pylint: disable=W0223
+    def __init__(self, policy: TFPolicy, trainer_params: Dict[str, Any]):
+        self.sess = policy.sess
+        self.policy = policy
```

Comment:
Slightly concerned about making our TFOptimizer base single policy centric.  Might need to build another class on top of Optimizer for multiagent training.

Reply:
Conclusion is to insert new single-agent and multi-agent classes between TFOptimizer and e.g. PPOOptimizer. Will be done in future PR.